scripts/test_translator.py

```diff
@@ -244,11 +244,20 @@ def __init__(self, full_path: str, files: 're.Pattern', keep: List[str], log_lev
         # are broken without it on macOS 12.
         # limit this to macOS because if we do happen to have multiple versions of Clang around, we
         # don't know which to use here, and using the wrong can one break things badly
-        if sys.platform == "darwin":
+        if sys.platform == "darwin" or os.environ.get("C2RUST_USE_NIX") == "1":
             _, stdout, _ = clang["-print-resource-dir"].run(retcode=None)
-            self.clang_resource_dir = " \"-I{}/include\",".format(stdout.strip())
+            self.clang_extra_flags = " \"-I{}/include\",".format(stdout.strip())
         else:
-            self.clang_resource_dir = ""
+            self.clang_extra_flags = ""
+
+        # The compile_commands.json needs to have extra flags to tell
+        # it where to find libraries when using nix.
+        if os.environ.get("C2RUST_USE_NIX") == "1":
+            nix_compile_flags = os.environ.get("BINDGEN_EXTRA_CLANG_ARGS", "") + os.environ.get("NIX_CFLAGS_COMPILE", "")
+            nix_compile_flags = nix_compile_flags.replace('-isystem ', '-I').replace('-idirafter ', '-I')
```

Comment:
I think we're not the most consistent already, but I think `"` double quotes are more commonly used.  Could you stick to them (except for things like `'"{}"'`)?

Reply:
Yep, no problem. I'll update my changes to prefer double quotes.